diff -urt flameshot.orig/src/utils/confighandler.cpp flameshot.new/src/utils/confighandler.cpp
--- flameshot.orig/src/utils/confighandler.cpp	2020-02-29 14:15:14.255261408 +0100
+++ flameshot.new/src/utils/confighandler.cpp	2020-02-29 14:25:34.081602795 +0100
@@ -77,15 +77,16 @@
 QVector<QColor> ConfigHandler::getUserColors() {
     QVector<QColor> colors;
     const QVector<QColor> &defaultColors = {
-        Qt::darkRed,
-        Qt::red,
-        Qt::yellow,
-        Qt::green,
-        Qt::darkGreen,
-        Qt::cyan,
-        Qt::blue,
-        Qt::magenta,
-        Qt::darkMagenta
+        QColor("#ffffff"),
+        QColor("#000000"),
+        QColor("#e7e6e6"),
+        QColor("#43a6d1"),
+        QColor("#ed7931"),
+        QColor("#1d7ea8"),
+        QColor("#387866"),
+        QColor("#75bd75"),
+        QColor("#e89929"),
+        QColor("#faad33")
     };
 
     if (m_settings.contains(QStringLiteral("userColors"))) {
Only in flameshot.new/src/utils: confighandler.cpp.orig
